codec.go

```diff
@@ -254,7 +256,11 @@ func (cdc *Codec) setTypeInfo_nolock(info *TypeInfo) {
 		if existing, ok := cdc.disfixToTypeInfo[disfix]; ok {
 			panic(fmt.Sprintf("disfix <%X> already registered for %v", disfix, existing.Type))
 		}
+		if existing, ok := cdc.nameToTypeInfo[info.Name]; ok {
+			panic(fmt.Sprintf("name <%s> already registered for %v", info.Name, existing.Type))
```

Comment:
Question: this does not panic in practice because we'd register with the "absolute name"? 

Different from the tests below https://github.com/tendermint/go-amino/pull/151/files#diff-28aea91bac84cd18870296d459932866R20; `car` would rather be `/unique/long/absolute/path/to/car`; is that right?

Reply:
In production code, it would be /unique/long/absolute/path/to/car, yes.  Maybe we should enforce some convention here, like having 2 parts like domain/name.